ComplementaryScripts/modelCuration/addMetabolomicsRxnToGEM.m

```diff
@@ -0,0 +1,149 @@
+    matrix.Newcomps(i) = strcat(llbracket,matrix.Newcomps(i),rbrackets);
+end
+
+% Map mets to model.metnames, get s_index for new mets:
+cd otherChanges
+for j = 1:length(matrix.metnames)
+    [~,metindex] = ismember(matrix.metnames(j),model.metNames);
+    if metindex ~= 0
+        matrix.mets(j) = model.mets(metindex);
+    elseif metindex == 0
+        newID = getNewIndex(model.mets);
+        matrix.mets(j) = strcat('s_',newID,matrix.Newcomps(j));
+        model = addMetabolite(model,char(matrix.mets(j)), ...
+            'metName',matrix.metnames(j));
+    end
+end
+
+
+% Add metabolite data:
+fid = fopen('../../ComplementaryData/modelCuration/Metabolomics_newRxnMetAnnotation.tsv');
+newmet_annot         = textscan(fid,'%s %s %s %s %s %s %s %s','Delimiter','\t','HeaderLines',1);
+newmet.metNames      = newmet_annot{1};
+newmet.metFormulas   = newmet_annot{2};
+newmet.metCharges    = cellfun(@str2num, newmet_annot{3});
+newmet.metKEGGID     = newmet_annot{5};
+newmet.metChEBIID    = newmet_annot{6};
+newmet.metMetaNetXID = newmet_annot{7};
+
+
+fclose(fid);
+for i = 1:length(newmet.metNames)
+    [~,metID] = ismember(newmet.metNames(i),model.metNames);
+    if metID ~= 0
+        model.metFormulas{metID}   = newmet.metFormulas{i};
+        model.metCharges(metID)    = newmet.metCharges(i);
+        model.metKEGGID{metID}     = newmet.metKEGGID{i};
+        model.metChEBIID{metID}    = newmet.metChEBIID{i};
+        model.metMetaNetXID{metID} = newmet.metMetaNetXID{i};
+        model.metNotes{metID}      = '';
+    end
+end
+
+% Add new reactions according to rev ID: Met Coef needs to be a column, not
+% a row. Coef should be a double, which was converted at the import section
+EnergyResults     = {};
+MassChargeresults = {};
+RedoxResults      = {};
+if ~isfield(model,'rxnMetaNetXID')
+    model.rxnMetaNetXID = cell(size(model.rxns));
+end
+for i = 1:length(newrxn.ID)
+    newID = getNewIndex(model.rxns);
+    j     = find(strcmp(matrix.rxnIDs,newrxn.ID{i}));
+    Met   = matrix.mets(j);
+    Coef  = transpose(matrix.metcoef(j));
+    [model,rxnIndex] = addReaction(model, ['r_' newID],...
+        'reactionName', newrxn.ID{i},...
+        'metaboliteList',Met,...
+        'stoichCoeffList',Coef,...
+        'reversible',newrxn.Rev(i,1),...
+        'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
+    if isempty(rxnIndex)
+        rxnIndex = strcmp(model.rxns,['r_' newID]);
+    end
+    % Add rxn annotation:
+    model.rxnNames{rxnIndex}      = newrxn.rxnNames{i};
+    model.rxnECNumbers(rxnIndex)  = newrxn.rxnECNumbers(i);
+    model.rxnKEGGID(rxnIndex)     = newrxn.rxnKEGGID(i);
+    model.rxnMetaNetXID(rxnIndex) = newrxn.rxnMetaNetXID(i);
+    model.rxnConfidenceScores(rxnIndex) = 1;   %reactions without gene but needed for modelling
```

Comment:
@feiranl thanks for the changes, it's looking very good! last thing I noticed: Not sure why the confidence coefficients should be =1, as these new rxns are not really needed for modeling; I would instead assign a 0.

I'm also thinking that it could be useful to leave a record of why these rxns and mets were added, if later someone else comes across any of these elements. How about something like the following in the metabolite loop:
```
model.metNotes{metID} = 'added from metabolomics data';
```
And in the rxn loop:
```
model.rxnNotes{rxnIndex} = 'metabolites observed in metabolomics data';
```